spec/enumerables_spec.rb

```diff
@@ -0,0 +1,242 @@
+    end
+  end
+
+  describe '#my_any?' do
+    it 'return true if any of the array items are true when no block or argument given' do
+      expect(array.my_any?).to eq(array.any?)
+    end
+
+    it 'return true if any of the array items are member of such class' do
+      expect(array.my_any?(Numeric)).to eq(array.any?(Numeric))
+    end
+
+    it 'return true if any of the array items matches regex given' do
+      expect(array.my_any?(rg)).to eq(array.any?(rg))
+    end
+
+    it 'return true if any of the array items matches the given value' do
+      expect(array.my_any?(3)).to eq(array.any?(3))
+    end
+
+    it 'return true if any of the range items are member of given class' do
+      expect(range.my_any?(Numeric)).to eq(range.any?(Numeric))
+    end
+
+    it 'return true if any of the range items matches the given value' do
+      expect(range.my_any?(3)).to eq(range.any?(3))
+    end
+  end
+
+  describe '#my_none?' do
+    it 'return true if none of the array items are true when no block or argument given' do
+      expect(array.my_none?).to eq(array.none?)
+    end
+
+    it 'return true if none of the array items are member of such class' do
+      expect(array.my_none?(Numeric)).to eq(array.none?(Numeric))
+    end
+
+    it 'return true if none of the array items matches regex given' do
+      expect(array.my_none?(rg)).to eq(array.none?(rg))
+    end
+
+    it 'return true if none of the array items matches the given value' do
+      expect(array.my_none?(3)).to eq(array.none?(3))
+    end
+
+    it 'return true if none of the range items are member of given class' do
+      expect(range.my_none?(Numeric)).to eq(range.none?(Numeric))
+    end
+
+    it 'return true if none of the range items matches the given value' do
+      expect(range.my_none?(3)).to eq(range.none?(3))
+    end
+  end
+
+  describe '#my_count' do
+    it 'returns the number of items in the array through enumeration' do
+      expect(array.my_count).to eq(array.count)
+    end
+
+    it 'returns the number of items in the range through enumeration' do
+      expect(range.my_count).to eq(range.count)
+    end
+
+    it 'returns the number of items in the range through enumeration' do
+      expect(array.my_count(&block)).to eq(array.count(&block))
+    end
+
+    it 'returns the number of items in the array through enumeration' do
+      expect(array.my_count(3)).to eq(array.count(3))
+    end
+
+    it 'returns the number of items in the range through enumeration' do
+      expect(range.my_count(3)).to eq(range.count(3))
+    end
+  end
+
+  describe '#my_map' do
+    it 'returns an Enumerator if no block is given' do
+      expect(array.my_map).to be_a(Enumerator)
+    end
+
+    it 'returns an Enumerator if no block is given' do
+      expect(range.my_map).to be_a(Enumerator)
+    end
+
+    it 'returns a new array after block executed for each item' do
+      expect(array.my_map(&block)).to eq(array.map(&block))
+    end
```

Comment:
How are you checking it is in fact a new array?? You should check the original and see if it hasn't been mutated.